homework/src/app.js

```diff
@@ -8,4 +8,145 @@
+  array1 = str.split('');
+  array2 = array1.reverse();
+  string = array2.join('');
+  console.log(string);
+  
+  
+  
+  
+  
+  
+  //Arrays
+  const favoriteAnimals = ["blowfish", "capricorn", "giraffe"];
+  //Add a statement that adds your favorite animal say 'turtle' to the existing array.
+  favoriteAnimals.push('turtle');
+  //my new array will be ["blowfish", "capricorn", "giraffe", "turtle"]
+  
+  //Log your new array.
+  console.log(favoriteAnimals);
+  
+  //Now add another favorite animal to the array but make sure it is placed after 'blowfish' and before 'capricorn'.
+  favoriteAnimals.splice(1, 0, 'dog');
+  //my new array will be now ["blowfish", "dog", "capricorn", "giraffe", "turtle"]
+  
+  //Log your new array.
+  console.log(favoriteAnimals);
+  
+  //Delete 'giraffe' from the array.
+  favoriteAnimals.splice(3, 1);
+  //the array will be ["blowfish", "dog", "capricorn", "turtle"]
+  console.log(favoriteAnimals)
+  
+  
+  
+  
+  
+  
+  //QUESTIONS FROM HYF REPOS
+  
+  
+  //8.1 First declare at least four variables and assign them different data types.
+  let firstItem = 0;
+  let secItem = "HackYourFuture";
+  let thirdItem = 99.99;
+  let fourthItem = true;
+  
+  //8.2 For each variable write a console.log statement that logs the value
+  console.log("The value of my variable firstItem is: " + 0);
+  console.log("The value of my variable sectItem is: " + "HackYourFuture");
+  console.log("The value of my variable thirdItem is: " + 99.99);
+  console.log("The value of my variable fourthItem is: " + true);
+  
+  //8.3 Now write a console.log statement wherein you first explain in words what you think the type of your variables is.
+  console.log("the typeof firstItem is : number");
+  console.log("the typeof secItem is : string");
+  console.log("the typeof thirdtItem is : number");
+  console.log("the typeof fourthItem is : boolean");
+  
+  //8.4 Now use typeof to log the actual type of your variables.
+  console.log(typeof firstItem);
+  console.log(typeof secItem);
+  console.log(typeof thirdItem);
+  console.log(typeof fourthItem);
+  
+  //8.5 Now compare the types of your different variables with one another.
+  if (typeof firstItem == typeof secItem) {
+    console.log('SAME TYPE');
+  }
+  else {
+    console.log('NOT SAME TYPE');
+  
+  }
+  
+  if (typeof firstItem == typeof thirdItem) {
+    console.log('SAME TYPE');
+  }
+  else {
+    console.log('NOT SAME TYPE');
+  
+  }
+  
+  
+  //9. If x equals 7, and the only other statement is x = x % 3, what would be the new value of x?
+  let x = 7;
+  x = x % 3;
+  console.log(x) // the value of x is 1
+  
+  
+  //9.1 Add at least 3 console.log statements in which you show that you understand what % does.
+  let y = 9;
+  y = y % 3;
+  console.log(y); //the value will be 0
+  
+  
+  let z = 15;
+  z = z % 2;
+  console.log(z); //the value will be 1
+  
+  
+  
+  let num = 22;
+  num = num % 10;
+  console.log(num); // the value will be 2
+  
+  
```

Comment:
Correct. Same remark as with others whereby you use the same variable to hold a value and to hold the outcome of an operation: num = num % 10; instead of let newNum = num % 10; I would do that since it lokos less confusing. JavaScript finds this ok (and maybe Yash too so then you're safe ;))

Reply:
yes,right! I will change it anyway since it is safer :)